enterprise/server/remote_execution/copy_on_write/copy_on_write.go

```diff
@@ -1213,8 +1223,19 @@ func NewMmapLRU() (*MmapLRU, error) {
 			if reason == lru.ManualEviction {
 				return
 			}
-			if err := m.unmap(true /*=isLRUEviction*/); err != nil {
-				log.Errorf("Failed to unmap chunk: %s", err)
+			// Unmap in the background to prevent a deadlock here in the case
+			// where the mmap we're evicting is already locked by another
+			// goroutine, and will make a call to lru.Add or lru.Remove before
+			// releasing its lock (the deadlock happens because we're already
+			// holding the LRU lock here).
+			//
+			// If the evictors are not keeping up, start our own goroutine to
+			// ensure the eviction still happens.
+			select {
+			case ml.evictions <- m:
+			default:
+				log.Warningf("LRU evictors are not keeping up")
+				go ml.processEviction(m)
 			}
```

Comment:
I wonder if blocking here would be better than starting another goroutine? Basically as a way to apply backpressure, rather than starting a bunch of goroutines that then might also be backed up.

Reply:
We can't block here because then it could lead to the deadlock scenario again.

I will add an UnexpectedEvent here instead of a Warningf since I don't think we'll hit the channel limit in practice